core/ledger/confighistory/mgr_test.go

```diff
@@ -300,100 +296,210 @@ func newTestEnvForSnapshot(t *testing.T) *testEnvForSnapshot {
+			info: &common.BlockchainInfo{Height: 1000},
+			qe:   &mock.QueryExecutor{},
+		}
+
+		retriever = env.mgr.GetRetriever("ledger3", dummyLedgerInfoRetriever)
+		verifyImportedConfigHistory(t, retriever, ccConfigInfo)
+		cleanup()
+	})
+
+	t.Run("export from an imported confighistory", func(t *testing.T) {
+		// setup ledger4 => export ledger4 => import into ledger5 => export ledger5
+		storedKVs, _ := setup("ledger4")
+		retriever := env.mgr.GetRetriever("ledger4", nil)
+		_, err := retriever.ExportConfigHistory(env.testSnapshotDir, testNewHashFunc)
+		require.NoError(t, err)
+
+		importConfigsBatchSize = 100
+		require.NoError(t, env.mgr.ImportConfigHistory("ledger5", env.testSnapshotDir))
+		cleanup()
+
+		retriever = env.mgr.GetRetriever("ledger5", nil)
+		fileHashes, err := retriever.ExportConfigHistory(env.testSnapshotDir, testNewHashFunc)
+		require.NoError(t, err)
+		verifyExportedConfigHistory(t, env.testSnapshotDir, fileHashes, storedKVs)
+		cleanup()
+	})
+
+	t.Run("import confighistory error cases", func(t *testing.T) {
+		setup("ledger6")
+		retriever := env.mgr.GetRetriever("ledger6", nil)
+		_, err := retriever.ExportConfigHistory(env.testSnapshotDir, testNewHashFunc)
+		require.NoError(t, err)
+
+		err = env.mgr.ImportConfigHistory("ledger6", env.testSnapshotDir)
+		expectedErrStr := "config history for ledger [ledger6] exists. Incremental import is not supported. Remove the existing ledger data before retry"
+		require.EqualError(t, err, expectedErrStr)
+
+		require.NoError(t, os.RemoveAll(filepath.Join(env.testSnapshotDir, snapshotDataFileName)))
+		err = env.mgr.ImportConfigHistory("ledger8", env.testSnapshotDir)
+		require.Contains(t, err.Error(), "confighistory.data: no such file or directory")
+
+		dataFileWriter, err := snapshot.CreateFile(filepath.Join(env.testSnapshotDir, snapshotDataFileName), snapshotFileFormat, testNewHashFunc)
+		require.NoError(t, err)
+		defer dataFileWriter.Close()
+		require.NoError(t, dataFileWriter.EncodeUVarint(1))
+		err = env.mgr.ImportConfigHistory("ledger8", env.testSnapshotDir)
+		require.Contains(t, err.Error(), "error while reading from the snapshot file")
+		require.Contains(t, err.Error(), "confighistory.data: EOF")
+
+		require.NoError(t, os.RemoveAll(filepath.Join(env.testSnapshotDir, snapshotMetadataFileName)))
+		err = env.mgr.ImportConfigHistory("ledger8", env.testSnapshotDir)
+		require.Contains(t, err.Error(), "confighistory.metadata: no such file or directory")
+
+		dataFileWriter, err = snapshot.CreateFile(filepath.Join(env.testSnapshotDir, snapshotMetadataFileName), snapshotFileFormat, testNewHashFunc)
+		require.NoError(t, err)
+		defer dataFileWriter.Close()
+		require.NoError(t, dataFileWriter.EncodeBytes([]byte("junk")))
+		err = env.mgr.ImportConfigHistory("ledger8", env.testSnapshotDir)
+		require.Contains(t, err.Error(), "error while reading from the snapshot file")
+		require.Contains(t, err.Error(), "confighistory.metadata: EOF")
+		cleanup()
+	})
```

Comment:
Can you add to this list one more test when the db returns an error while importing

Reply:
I added it initially to check the db writebatch. Once I added error for the `iter.Next()`, I couldn't test the writebatch error. I, in fact, made the `maxBatchSize` as var instead of constant so that the test could covered both writebatches (inside and outside the loop). I will make it constant and add a test for the db error (but it would return at `isEmpty()` check itself)

Reply:
I understand that it's hard to skip isEmpty but whatever can be added to test, is good.